SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs

```diff
@@ -0,0 +1,129 @@
+            // return _context.Competitions.ToArray<Competition>();
+            return competitionsIDs;
+        }
+//----------------------------------------------------------------------------------------------------------------------------------------
+        //Getting the competition by ID
+        // GET: api/Competition/5
+        [HttpGet("{id}")]
+        public async Task<Competition> CompetitionGet(int id)
+        {
+            return await _context.Competitions.FindAsync(id);
+        }
+ //----------------------------------------------------------------------------------------------------------------------------------------
+        //Getting the competition by ID
+        // GET: api/Competition/5
+        [HttpGet("{Name}")]
+        public async Task<Competition> CompetitionGet(string Name)
+        {
+            return await _context.Competitions.FindAsync(Name);
+        }
+//----------------------------------------------------------------------------------------------------------------------------------------
+        //posting the competition to the competition table
+        // POST: api/Competition
+        [HttpPost]
+        public async Task<IActionResult> addCompetition([FromBody]Competition comp)
+        {
+            if (ModelState.IsValid)
+            {
+                Competition dbComp = null; //assume the competition does not exist
+
+                dbComp = _context.Competitions.FirstOrDefault(x => x.Name == comp.Name);
+                //if competition aready exist return
+                if (dbComp != null)
+                {
+                    return Ok("User already exists");
+                }
+                //Save the competition basic details
+                await _context.AddAsync(comp);
+                await _context.SaveChangesAsync();
+
+                return Ok("Competition saved successfully");
+            }
+            else
+            {
+                return new BadRequestObjectResult("competition cannot be null");
+            }
+        }
+//----------------------------------------------------------------------------------------------------------------------------------------
+        //A method that updates the status of the competition
+        // PUT: api/Competition/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> UpdateCompetition(int id, [FromBody]Competition comp)
+        {
+            if (ModelState.IsValid)
+            {
+                //error handling, check if client provided valid data
+                if (comp == null)
+                {
+                    return new BadRequestObjectResult("competiton cannot be null");
+                }
+                else
+                {
+                    Competition dbComp = null; //assume competition does not exist
+                    using (_context)
+                    {
+                        dbComp = _context.Competitions
```

Comment:
I would change this to be a boolean with an `Any` query. This is a new username, so you only care about the first occurrence of this username. Your current statement will iterate all the records regardless if it has been found or not. Using an Any() will stop the search once a matching username was found, increasing the performance.